packages/edit-site/src/components/actions/index.js

```diff
@@ -10,62 +10,90 @@ import { __, sprintf } from '@wordpress/i18n';
 import { store as noticesStore } from '@wordpress/notices';
 import { useMemo } from '@wordpress/element';
 import { privateApis as routerPrivateApis } from '@wordpress/router';
+import {
+	Button,
+	__experimentalText as Text,
+	__experimentalHStack as HStack,
+	__experimentalVStack as VStack,
+} from '@wordpress/components';
+
 /**
  * Internal dependencies
  */
 import { unlock } from '../../lock-unlock';
 
 const { useHistory } = unlock( routerPrivateApis );
 
-export function useTrashPostAction() {
-	const { createSuccessNotice, createErrorNotice } =
-		useDispatch( noticesStore );
-	const { deleteEntityRecord } = useDispatch( coreStore );
-
-	return useMemo(
-		() => ( {
-			id: 'move-to-trash',
-			label: __( 'Move to Trash' ),
-			isPrimary: true,
-			icon: trash,
-			isEligible( { status } ) {
-				return status !== 'trash';
-			},
-			async callback( post ) {
-				try {
-					await deleteEntityRecord(
-						'postType',
-						post.type,
-						post.id,
-						{},
-						{ throwOnError: true }
-					);
-					createSuccessNotice(
-						sprintf(
-							/* translators: The page's title. */
-							__( '"%s" moved to the Trash.' ),
-							decodeEntities( post.title.rendered )
-						),
-						{
-							type: 'snackbar',
-							id: 'edit-site-page-trashed',
-						}
-					);
-				} catch ( error ) {
-					const errorMessage =
-						error.message && error.code !== 'unknown_error'
-							? error.message
-							: __(
-									'An error occurred while moving the page to the trash.'
-							  );
+export const trashPostAction = {
+	id: 'move-to-trash',
+	label: __( 'Move to Trash' ),
+	isPrimary: true,
+	icon: trash,
+	isEligible( { status } ) {
+		return status !== 'trash';
+	},
+	hideModalHeader: true,
+	RenderModal: ( { item: post, closeModal } ) => {
```

Comment:
In essence the changes are:
1. add `hideModalHeader` and `RenderModal`
2. `RenderModal` absorbs the `callback`
3. use a plain object instead of a hook.